kong/plugins/opentelemetry/handler.lua

```diff
@@ -1,211 +1,55 @@
-  })
-  if not res then
-    return false, "failed to send request: " .. err
-
-  elseif res and res.status ~= 200 then
-    return false, "response error: " .. tostring(res.status) .. ", body: " .. tostring(res.body)
-  end
-
-  return true
-end
-
-
-local function http_export(conf, spans)
-  local start = ngx_now()
-  local headers = get_headers(conf.headers)
-  local payload = encode_traces(spans, conf.resource_attributes)
-
-  local ok, err = http_export_request(conf, payload, headers)
-
-  ngx_update_time()
-  local duration = ngx_now() - start
-  ngx_log(ngx_DEBUG, _log_prefix, "exporter sent ", #spans,
-          " traces to ", conf.endpoint, " in ", duration, " seconds")
-
-  if not ok then
-    ngx_log(ngx_ERR, _log_prefix, err)
-  end
-
-  return ok, err
-end
-
 
-local function get_inject_ctx(extracted_ctx, conf)
-  local root_span = ngx.ctx.KONG_SPANS and ngx.ctx.KONG_SPANS[1]
-
-  -- get the global tracer when available, or instantiate a new one
-  local tracer = kong.tracing.name == "noop" and kong.tracing.new("otel")
-                 or kong.tracing
-
-  -- make propagation work with tracing disabled
-  if not root_span then
-    root_span = tracer.start_span("root")
-    root_span:set_attribute("kong.propagation_only", true)
-
-    -- since tracing is disabled, turn off sampling entirely for this trace
-    kong.ctx.plugin.should_sample = false
-  end
-
-  local injected_parent_span = tracing_context.get_unlinked_span("balancer") or root_span
-  local trace_id = extracted_ctx.trace_id
-  local span_id = extracted_ctx.span_id
-  local parent_id = extracted_ctx.parent_id
-  local parent_sampled = extracted_ctx.should_sample
-
-  -- Overwrite trace ids
-  -- with the value extracted from incoming tracing headers
-  if trace_id then
-    -- to propagate the correct trace ID we have to set it here
-    -- before passing this span to propagation
-    injected_parent_span.trace_id = trace_id
-    -- update the Tracing Context with the trace ID extracted from headers
-    tracing_context.set_raw_trace_id(trace_id)
-  end
-
-  -- overwrite root span's parent_id
-  if span_id then
-    root_span.parent_id = span_id
-
-  elseif parent_id then
-    root_span.parent_id = parent_id
-  end
-
-  -- Configure the sampled flags
-  local sampled
-  if kong.ctx.plugin.should_sample == false then
-    sampled = false
-
-  else
-    -- Sampling decision for the current trace.
-    local err
-    -- get_sampling_decision() depends on the value of the trace id: call it
-    -- after the trace_id is updated
-    sampled, err = tracer:get_sampling_decision(parent_sampled, conf.sampling_rate)
-    if err then
-      ngx_log(ngx_ERR, _log_prefix, "sampler failure: ", err)
+function OpenTelemetryHandler:configure(configs)
+  if configs then
+    for _, config in ipairs(configs) do
+      if config.logs_endpoint then
+        dynamic_hook.hook("observability_logs", "push", o11y_logs.maybe_push)
+        dynamic_hook.always_enable("observability_logs")
```

Comment:
After reconfiguring, did we disable the `observability_logs`?

Reply:
should we add a feature to make it possible to disable dynamic hooks? Unless I missed it that doesn't seem possible at the moment. Easy enough to do anyway, I already have a commit that makes a small change there, I can squash that together.

Reply:
I think it should be easy to do, let's try two interfaces:

```lua
dynamic_hook.enable_by_default(group_name)
dynamic_hook.disable_by_default(group_name)
```

And remove the interface `dynamic_hook.always_enable`

Reply:
done.

Reply:
done